iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergQueryCompactor.java

```diff
@@ -88,28 +89,30 @@ public boolean run(CompactorContext context) throws IOException, HiveException,
         // added another filter on file_path as a workaround.
         compactionQuery = String.format("insert overwrite table %1$s select * from %1$s " +
                 "where %2$s != %3$d and %4$s is not null %5$s %6$s",
-            compactTableName, VirtualColumn.PARTITION_SPEC_ID.getName(), icebergTable.spec().specId(),
+            qualifiedTableName, VirtualColumn.PARTITION_SPEC_ID.getName(), icebergTable.spec().specId(),
             VirtualColumn.FILE_PATH.getName(), fileSizePredicate == null ? "" : "and " + fileSizePredicate, orderBy);
       } else {
         // Partitioned table without partition evolution with partition spec as null in the compaction request - this
         // code branch is not supposed to be reachable
         throw new HiveException(ErrorMsg.COMPACTION_NO_PARTITION);
       }
     } else {
-      long partitionHash = IcebergTableUtil.getPartitionHash(icebergTable, partSpec);
+      int partitionSpecId = IcebergTableUtil.getPartitionSpecId(icebergTable, partSpec);
       HiveConf.setVar(conf, ConfVars.REWRITE_POLICY, RewritePolicy.PARTITION.name());
       conf.set(IcebergCompactionService.PARTITION_PATH, new Path(partSpec).toString());
 
       Map<String, String> partSpecMap = new LinkedHashMap<>();
       Warehouse.makeSpecFromName(partSpecMap, new Path(partSpec), null);
 
-      compactionQuery = String.format("insert overwrite table %1$s select * from %1$s where %2$s=%3$d " +
-              "and %4$s is not null %5$s %6$s", compactTableName, VirtualColumn.PARTITION_HASH.getName(), partitionHash,
-          VirtualColumn.FILE_PATH.getName(), fileSizePredicate == null ? "" : "and " + fileSizePredicate, orderBy);
+      compactionQuery = String.format("INSERT OVERWRITE TABLE %1$s SELECT * FROM %1$s " +
+              "WHERE %2$s LIKE '%%/%3$s/data/%4$s/%%' AND %5$s = %6$d %7$s %8$s",
```

Comment:
`'%%/%3$s/data/%4$s/%%'` assumes a strict file structure. In case that users performed the in-place migration, the file name can't follow the convention. It sounds acceptable at this point.

Reply:
Improved the fix to not assume that files are places under data folder.